### PR TITLE
Default to logged in user if none specified, reactively.

### PR DESCRIPTION
**Pull request by trusktr**
Solves #53

### CODE REVIEW

File: export.js [hunk 1]
```diff
@@ -135,6 +135,14 @@ Avatar = {
   // Get the url of the user's avatar
   getUrl: function (user) {
 
+    // This is reactive, which causes Avatar.getUrl to trigger reactivity when
+    // it is used in reactive computations. There's no harm done to code that
+    // uses this function in non-reactive places.
+    var _user = Meteor.user()
```
Comment (timothyarmes) [inline export.js]: This code will cause a re-render for every change to the Meteor.user() object even if there's another user specified for the Avatar. That's pointless. Instead you should simply do:

if (!user) user = Meteor.user()

This also avoids the _ prefix which doesn't follow the coding style of the rest of the code base :)

File: blaze/avatar.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ Template.avatar.helpers({
 
   imageUrl: function () {
     var user = this.user ? this.user : Meteor.users.findOne(this.userId);
```
Comment (timothyarmes) [inline blaze/avatar.js]: If user becomes optional then it's worth avoiding the findOne() when there's no userId specified

**Comment by trusktr** [comment]
Good points! We only care about userId changing, not all the user's data. I've updated it. Is that what you meant by avoiding the findOne call?

File: export.js [hunk 3]
```diff
@@ -135,6 +135,9 @@ Avatar = {
   // Get the url of the user's avatar
   getUrl: function (user) {
 
+    // Default to the currently logged in user, unless otherwise specified.
+    if (!user) user = Meteor.users.findOne(Meteor.userId())
```
Comment (timothyarmes) [inline export.js]: I still think you need this:

```
if (!user) user = Meteor.user()
```

Otherwise it won't update if a custom avatar URL changes.

Comment (trusktr) [inline export.js]: @timothyarmes Hmm, good point again! But, will this mean that renders (when a user isn't supplied to `Avatar.getUrl`) will happen a lot more often if anything in the `Meteor.user()` object changes, even if not related to the avatar?

Comment (timothyarmes) [inline export.js]: Yes, but just for the logged on user. Most avatars will be for other users.

Comment (trusktr) [inline export.js]: @timothyarmes There we go. How does that look?

**Pull request merged**